[ADD] l10n_ar_edi_ux, l10n_ar_currency_update: Re use AFIP connection

zaoral opened the pull request:
**Update From Padron AFIP** and "**Currency rate sincronization** are only working if:

* my logged company has a valid certificate and connection to ws_sr_constancia_inscripcion/wsbe webservice
* my logged company has not certificate and there is only ONE other company with certificate and a valid connection to the
  wsbe/ws_sr_constancia_inscripcion webservices.

This PR changes the logic to always try to find a company that actually can connect to the webserive and use that one in order
to connect to AFIP, avoiding trying to connect with an invalid connection which results on not working padron an rate syncronization (mostly silence errors)

zaoral committed:
[ADD] l10n_ar_currency_update: find company with valid connection

Improves the company selection logic for AFIP currency updates by not
only checking for valid certificates but also verifying that companies
have valid connections to the required web services.

* Extracts certificate validation logic into a reusable method
  _find_company_with_valid_certificate
* Adds web service connection validation to ensure companies can
  actually connect to AFIP services
* FIX: Only ask this once outsite for loop

roboadhoc commented:
[![Pull request status dashboard](https://mergebot.adhoc.ar/ingadhoc/odoo-argentina-ee/pull/681.png)](https://mergebot.adhoc.ar/ingadhoc/odoo-argentina-ee/pull/681)

zaoral commented on l10n_ar_currency_update/models/res_company.py:
```diff
@@ -75,15 +98,7 @@ def _parse_afip_data(self, available_currencies):
         rate_date = today
 
```
company = self._find_company_with_valid_certificate(today, "wsfe")

zaoral commented on l10n_ar_currency_update/models/res_company.py:
```diff
@@ -75,15 +98,7 @@ def _parse_afip_data(self, available_currencies):
         rate_date = today
 
         for currency in available_currencies:
-            valid_certificate = (
```
seudo

```python
if metodo xxx exite:
   company = llamar metodo
else:
    codigo actual de compania
```

zaoral committed:
[ADD] l10n_ar_edi_ux: find company with valid connection

Update From Padron AFIP is only working when

* my login company has a valid certificate and connection to
  ws_sr_constancia_inscripcion webservice
* my login company has not certificate and there is only one other
  company with certificate and connection to the
  ws_sr_constancia_inscripcion webservice.

This PR changes the logic to always try to find a company that
actually can connect to the webserive and use that one in order
to connect to AFIP to get padron data.
